pkg/common/health/health.go

```diff
@@ -0,0 +1,95 @@
+package health
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/InVisionApp/go-health"
+	"github.com/InVisionApp/go-health/handlers"
+	"github.com/sirupsen/logrus"
+)
+
+// health.Checker is responsible for running health checks and serving the healthcheck HTTP paths
+type Checker struct {
+	config Config
+
+	server *http.Server
+
+	hc *health.Health
+
+	log logrus.FieldLogger
+}
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusOK)
+}
+
+func NewChecker(config Config, log logrus.FieldLogger) *Checker {
+	hc := health.New()
+
+	var server http.Server
+	// Start HTTP server if address is configured
+	if config.Address != "" {
+		handler := http.NewServeMux()
+
+		handler.HandleFunc(config.getHealthPath(), handlers.NewJSONHandlerFunc(hc, nil))
+		handler.HandleFunc(config.getLivePath(), hello)
+
+		server = http.Server{
+			Addr:    config.Address,
+			Handler: handler,
+		}
+	}
+
+	hc.StatusListener = &statusListener{}
+
+	return &Checker{config: config, server: &server, hc: hc, log: log}
+}
+
+func (c *Checker) AddCheck(name string, checker health.ICheckable) error {
+	return c.hc.AddCheck(&health.Config{
+		Name:     name,
+		Checker:  checker,
+		Interval: c.config.getInterval(),
+		Fatal:    true,
+	})
+}
+
+func (c *Checker) ListenAndServe(ctx context.Context) error {
+	if err := c.hc.Start(); err != nil {
```

Comment:
Enabling health check listener should be optional.

Given util.RunTasks consideration, perhaps we don't bother setting up any of the health.New() stuff in NewChecker in the event that we are disabled, and AddCheck and ListenAndServe return nil?

Reply:
If no address is specified, it doesn't listen -- it'll still log healthy/unhealthy state transitions though, which seemed valuable to me.  I don't have strong opinions on what the level of configiness should be here.

Reply:
Aah gotcha sorry I missed the conditional below. Yes logging health state changes seems valuable regardless of whether the listener is enabled or not - thanks for pointing that out.

Reply:
It's explicitly configurable now